### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "original" argument must be of type function

### ISSUE THREAD

**Issue by kouohhashi** [thenativeweb/crypto2#4]
After upgrading to 1.0.2, I start seeing the error, "TypeError [ERR_INVALID_ARG_TYPE]: The "original" argument must be of type function". Thanks!

**Comment by goloroden** [comment]
Sorry for not having responded earlier. Can you please give a [short, but complete program](http://jonskeet.uk/csharp/complete.html) to reproduce this issue?

@grundhoeferj Could you please have a look at this issue?

**Comment by Schwartz10** [comment]
I'm getting the same.. All you have to do to recreate the bug is require crypto2 in the front end of a react app

**Comment by tasoskalogero** [comment]
Same error here.  
`const crypto2 = require('crypto2');` causes the error..

**Comment by goloroden** [comment]
Sorry for not having responded earlier. I am having troubles recreating the problem 😞 

My sample app looks like this:

```javascript
'use strict';

const crypto2 = require('crypto2');

crypto2.hash('the native web').
  then(hash => console.log(hash));
```

This runs with `crypto2` in version `1.0.2` without further ado on Node.js `8.9.4` and `6.11.4`. I assume that this does *not* work for you, does it?

**Comment by kouohhashi** [comment]
```
create-react-app test-app
cd test-app
npm install --save crypto2
npm install
npm run start
```

And I can see the error.
create-react-app may cause some problem...

Here's my dependancy.
```
  "dependencies": {
    "crypto2": "^1.0.2",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "react-scripts": "^1.1.1"
  },
```
Node version: 
v8.9.1

Thanks.

**Comment by goloroden** [comment]
Okay, I found the problem… the reason for this behavior is that internally, `crypto2` [makes use of the `fs.readFile` function](https://github.com/thenativeweb/crypto2/blob/5ecb76f49e358323f0102c20ad7e5140d0eb49c5/lib/crypto2.js#L11), which of course doesn't exist in the browser, hence is `undefined`, and that's why the call to the `promisify` function fails.

So, the question is: How should we deal with this? Any suggestions?

**Comment by gagathos** [comment]
@goloroden I have solved the issue (I think) but I cannot figure out the build process to get a proper dist (I get incompatible license errors)

https://github.com/gagathos/crypto2

The change is simple: `readFile = typeof fs.readFile === 'function' ? promisify(fs.readFile) : {};`

**Comment by goloroden** [comment]
@gagathos Thanks for bringing up this idea 😊 

Since `readFile` should be a function in the end, I think we should update this to something such as:

```javascript
readFile = typeof fs.readFile === 'function' ? promisify(fs.readFile) : function () {};
```

And, since it should be a function which returns a promise, but it doesn't work in the browser, what about something like this:

```javascript
readFile = typeof fs.readFile === 'function' ?
  promisify(fs.readFile) :
  async function () {
    throw new Error('Invalid operation.');
  };
```

Would you mind turning this code into a PR? Don't worry regarding the build, I can then do this.

**Comment by goloroden** [comment]
Solved by 071d3b6c610857cf5df28fd3727e77b2ce6a47a9
